gnd/src/main/java/com/google/android/gnd/persistence/sync/LocalMutationSyncWorker.java

```diff
@@ -112,17 +113,26 @@ private Completable processMutations(ImmutableList<Mutation> mutations, String u
   private Completable processMutations(ImmutableList<Mutation> mutations, User user) {
     return remoteDataStore
         .applyMutations(mutations, user)
-        .andThen(deleteObservationIfRemovedRemotely(mutations))
+        .andThen(deleteObservationOrFeature(mutations))
         .andThen(localDataStore.removePendingMutations(mutations));
   }
 
   // TODO: If the remote sync fails, reset the state to DEFAULT.
-  private Completable deleteObservationIfRemovedRemotely(ImmutableList<Mutation> mutations) {
+  private Completable deleteObservationOrFeature(ImmutableList<Mutation> mutations) {
```

Comment:
The naming of this method is a bit misleading; it actually filters out delete mutations and applies them to the local db. To make this clearer, I think we could do something like the following:

```
  private Completable processMutations(ImmutableList<Mutation> mutations, User user) {
    return remoteDataStore
        .applyMutations(mutations, user)
        .andThen(localDataStore.finalizeMutations(mutations));
  }
```

Then from `LocalDataStore.finalizePendingMutations`  we could `finalizeDeletions` and `removePending`, both in `LocalDataStore`. Wdyt?

Reply:
Thanks! This makes so much sense now
05b7f5b